File: backend/app/src/models/buildings/GoldMine.py
from .Mine import Mine


class GoldMine(Mine):
    def __init__(self, workers: list = []) -> None:
        super().__init__(name="Gold Mine", worker_slots=2, workers=workers)
        
    
    def convert_worker_to_miner(self, worker, location) -> bool:
        if self.convert_worker_to_craftsman("Gold Mine", worker, location):
            return True

        return False








if __name__ == "__main__":
    pass

File: backend_refactoring/CSharpApi/Models/BluePrints/BuildingTypes/GoldMine.cs

using CSharpApi.Models.BluePrints.Beings;
using CSharpApi.Models.BluePrints.Locations;

namespace CSharpApi.Models.BluePrints.BuildingTypes
{
    public class GoldMine : Mine, IBuildingList
    {
        public string Name { get; set; } = "Gold Mine";

        public Dictionary<string, Dictionary<string, float>> Costs { get; set; } = new() {
            { "costs", new()
                {
                    { "gold", 200 },
                    { "food", 100 },
                    { "wood", 200 },
                    { "iron", 50 }
                }
            }
        };

        private List<Worker> _workers;

        public GoldMine(List<Worker> workers = null)
           : base("Gold Mine", true, 2, workers ?? [])
        {
            _workers ??= workers ?? [];
        }

        public bool ConvertWorkerToMiner(int workerId, Settlement location)
        {
            return (bool)ConvertWorkerToCraftsman(Name, workerId, location)["success"];
        }
    }
}
